modules/vactory_otp/src/Controller/OtpController.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Drupal\vactory_otp\Controller;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigManager;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Provides otp endpoints.
+ */
+class OtpController extends ControllerBase {
+
+  /**
+   * Time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * Otp Login Config.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * Constructs a new ApiFlagging object.
+   */
+  public function __construct(TimeInterface $time, ConfigManager $config_manager) {
+    $this->time = $time;
+    $this->config = $config_manager->getConfigFactory()->getEditable('vactory_otp.login_settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('datetime.time'),
+      $container->get('config.manager'),
+    );
+  }
+
+  /**
+   * Generate OTP.
+   */
+  public function generateOtp(Request $request, $value) {
+    $login_field = $this->config->get('login_field');
+    $phone_field = $this->config->get('phone_field');
+    $email_field = $this->config->get('email_field');
+    $canal = $this->config->get('canal');
+
+    $query = \Drupal::entityTypeManager()->getStorage('user')->getQuery();
+    $query->condition($login_field, trim($value));
```

Comment:
@ismail-void 
In Drupal 10 access checking must be explicitly specified on content entity queries
https://www.drupal.org/node/3201242
$query->accessCheck(FALSE);